### ISSUE TITLE
CGContextSetStrokeColorWithColor: invalid context 0x0

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-native#1417]
Seen in debug spew when running the test app on device:

May  4 17:45:56 John-Firebaughs-iPhone Mapbox GL[9832] <Error>: CGContextSetStrokeColorWithColor: invalid context 0x0. This is a serious error. This application, or a library it uses, is using an invalid context  and is thereby contributing to an overall degradation of system stability and reliability. This notice is a courtesy: please fix this problem. It will become a fatal error in an upcoming update.

Related to touching the geolocation icon.

**Comment by friedbunny** [comment]
This would be eliminated by #1222, which moves mostly to CALayer-based drawing (instead of Core Graphics). That PR is planned for beta 2, if everything pans out.

**Comment by friedbunny** [comment]
```
May 20 11:04:14 Hobbes Mapbox GL[5934] <Error>: CGBitmapContextInfoCreate: unable to allocate 1102107200 bytes for bitmap data
May 20 11:04:14 Hobbes Mapbox GL[5934] <Error>: CGContextSetStrokeColorWithColor: invalid context 0x0. This is a serious error. This application, or a library it uses, is using an invalid context  and is thereby contributing to an overall degradation of system stability and reliability. This notice is a courtesy: please fix this problem. It will become a fatal error in an upcoming update.
May 20 11:04:14 Hobbes Mapbox GL[5934] <Error>: CGContextSetFillColorWithColor: invalid context 0x0. This is a serious error. This application, or a library it uses, is using an invalid context  and is thereby contributing to an overall degradation of system stability and reliability. This notice is a courtesy: please fix this problem. It will become a fatal error in an upcoming update.
May 20 11:04:14 Hobbes Mapbox GL[5934] <Error>: CGContextSetLineWidth: invalid context 0x0. This is a serious error. This application, or a library it uses, is using an invalid context  and is thereby contributing to an overall degradation of system stability and reliability. This notice is a courtesy: please fix this problem. It will become a fatal error in an upcoming update.
May 20 11:04:14 Hobbes Mapbox GL[5934] <Error>: CGContextStrokeEllipseInRect: invalid context 0x0. This is a serious error. This application, or a library it uses, is using an invalid context  and is thereby contributing to an overall degradation of system stability and reliability. This notice is a courtesy: please fix this problem. It will become a fatal error in an upcoming update.
```

Got this after reenabling location services for the demo app, but can't get it to reproduce reliably.

**Comment by friedbunny** [comment]
Fixed by #1222.
